packages/babel-plugin-minify-simplify/src/index.js

```diff
@@ -124,6 +124,54 @@ module.exports = ({ types: t }) => {
         ],
       },
 
+      LogicalExpression(path) {
+        const left = path.get("left");
+        const right = path.get("right");
+        const operator = path.node.operator;
+
+        const AND = (input) => input === "&&";
+        const OR = (input) => input === "||";
+
+        const TRUTHY = (input) => {
+          const evalResult = input.evaluate();
+          return evalResult.confident && input.isPure() && evalResult.value;
+        };
+        const FALSY = (input) => {
+          // NaN and undefined are falsy
+          if (input.isIdentifier()) {
```

Comment:
Is this needed because the `.isPure()` method doesn't work for `NaN` and `undefined` yet, since `!undefined` and `!NaN` are `true`? Are there any other such literals?